src/Stryker.Core/Stryker.Core/Initialisation/ProjectFileReader.cs

```diff
@@ -104,24 +104,29 @@ private IAnalyzerResult SelectAnalyzerResult(IAnalyzerResults analyzerResults, s
         {
             if (targetFramework == null)
             {
-                return analyzerResults.First(e => e.TargetFramework != null);
+                var analyzerResult = analyzerResults.FirstOrDefault(e => e.TargetFramework is not null);
+
+                if (analyzerResult is null)
+                {
+                    throw new InputException("No analyzer result found and the targetFramework was null.");
+                }
+
+                return analyzerResult;
             }
 
             var analyzerResultForFramework = analyzerResults.SingleOrDefault(result => result.TargetFramework == targetFramework);
-            if (analyzerResultForFramework != null)
+            if (analyzerResultForFramework is not null)
             {
                 return analyzerResultForFramework;
             }
-            else
-            {
-                var firstAnalyzerResult = analyzerResults.First();
-                _logger.LogWarning(
-                    "The configured target framework '{0}' isn't available for this project. " +
-                    "It will be built against the first framework available " +
-                    "which is {1}.", targetFramework, firstAnalyzerResult.TargetFramework);
 
-                return firstAnalyzerResult;
-            }
+            var firstAnalyzerResult = analyzerResults.First();
```

Comment:
```suggestion
            var firstAnalyzerResult = analyzerResults.First(e => e.TargetFramework is not null);
```

We cannot test against a analyzer result that does not have a valid target framework.